src/http-api/utils.ts

```diff
@@ -122,6 +134,26 @@ function getResponseContentType(response: XMLHttpRequest | Response): ParsedMedi
     }
 }
 
+/**
+ * Parse a Retry-After header value and convert it into a relative delay in milliseconds.
+ * @see {@link https://www.rfc-editor.org/rfc/rfc9110#section-10.2.3-2}
+ * @throws Error if the provided value is not a valid Date HTTP header value
+ */
+export function parseRetryAfterMs(retryAfter: string): number {
+    if (/^\d+$/.test(retryAfter)) {
+        const ms = Number.parseInt(retryAfter) * 1000;
+        if (!Number.isFinite(ms)) {
+            throw new Error("numeric value is too large");
+        }
+        return ms;
+    }
+    const date = new Date(retryAfter);
+    if (date.toUTCString() !== retryAfter) {
+        throw new Error("value does not match Date HTTP header syntax");
+    }
```

Comment:
somewhat wondering if this bit is necessary, but I don't have a strong opinion

Reply:
I put this in lest future server developers return Retry-After with dates formatted in some other way and get surprised that they aren't supported.

Reply:
> I put this in lest future server developers return Retry-After with dates formatted in some other way and get surprised that they aren't supported.

I think the surprise would be that the js-sdk *does* support it (whilst other client implementations might not)?

Anyway, being conservative about what we accept makes sense to me. :+1: